core/server/worker/src/main/java/alluxio/worker/netty/WriteRequestContext.java

```diff
@@ -175,6 +186,13 @@ public void setPosToWrite(long posToWrite) {
   }
 
   /**
+   * @param done whether the SUCCESS or CANCEL response is sent
+   */
+  public void setDoneUnsafe(boolean done) {
+    mDone = done;
+  }
```

Comment:
Instead of having a separate value, should we just derive this from the other two values?

Reply:
in the write context, no other values to derive. These are the available states:
```java
     mRequest = request;
      mPosToQueue = 0;
      mPosToWrite = 0;
      mDone = false;
```